core/server/master/src/main/java/alluxio/master/file/meta/InodeTree.java

```diff
@@ -735,6 +735,46 @@ public long reinitializeFile(LockedInodePath inodePath, long blockSizeBytes, lon
     return file.getId();
   }
 
+  /**
+   * Locks a specific descendant of a particular {@link LockedInodePath}. It does not extend the
+   * {@link LockedInodePath}, it only locks the descendant.
+   *
+   * @param inodePath the root to start locking
+   * @param lockMode the lock type to use
+   * @param descendantUri the path to the descendant that we are locking
+   * @return  an {@link InodeLockList} representing the list of descendants that got locked as
+   * a result of this call.
+   * @throws FileDoesNotExistException if inode does not exist
+   */
+  public InodeLockList lockDescendant(LockedInodePath inodePath, LockMode lockMode,
+      AlluxioURI descendantUri) throws FileDoesNotExistException, InvalidPathException {
+    InodeLockList inodeGroup = new InodeLockList();
+    // Check if the descendant is really the descendant of inodePath
+    if (!PathUtils.hasPrefix(descendantUri.getPath(), inodePath.getUri().getPath())
+        || descendantUri.getPath().equals(inodePath.getUri().getPath())) {
+      throw new InvalidPathException(descendantUri.getPath() + "is not a valid descendant of "
+          + inodePath.getUri().getPath());
+    }
+
+    List<Inode<?>> nonPersistedInodes = new ArrayList<>();
+    List<Inode<?>> inodeList = new ArrayList<>(inodePath.getInodeList());
+    for (Inode<?> inode : inodeList) {
+      if (!inode.isPersisted()) {
+        nonPersistedInodes.add(inode);
+      }
+    }
+    // Lock from inodePath to the descendant
+    InodeLockList lockList = new InodeLockList();
+    TraversalResult traversalResult = traverseToInodeInternal(
+        PathUtils.getPathComponents(descendantUri.getPath()),
+        inodeList, nonPersistedInodes, lockList, lockMode, null);
+    if (traversalResult.mFound) {
+      return traversalResult.mLockList;
+    } else {
+      throw new InvalidPathException(descendantUri.getPath() + "path not found in traversal.");
```

Comment:
NIT: a space before "path"

Also, include `inodePath.getUri().getPath()` in the error message.